new-frontiers-14/frontier-station-14
# Complete rewrite of latejoin window

## About the PR


Redesign and complete rewrite of the latejoin menu.
Added more descriptions for options, described what each station purpose is, can easily be changed. The new code allows adding a icon , small description, and long description, and a sorting priority to a station.

For downstream i made it easy to skip the first large two button screen if they want and go directly to station or crew tab.
Search functionality is removed for now and will be reimplemented at a later time.

## Why / Balance

The intent is to make new players to understand what they're getting into when pressing buttons by adding a small and long description for each station.

## How to test

insert commands:
forcemap Frontier
setgamepreset Adventure
golobby
startround

1. Start a game with latejoin lobby enabled.
2. Press join button
3. Click around things and join various jobs
4. Buy a ship on one character and add some jobs to the station records computer
5. Join stations and ships to test.

Some new edge cases:
- If there are no stations where player can spawn on, the station button will show "There are no stations available at this time." 
- Same goes for crews.

Menus are designed to be recycleable.
Code has good separation of concerns.
Please note that the crew control code is going to be changed later on so it is a copy of the station control.

## Media

![image](https://github.com/user-attachments/assets/ef9a174a-0bb7-4f59-87e8-36970e4d120a)
![image](https://github.com/user-attachments/assets/e3e52ba7-a36e-4bd3-9e0b-038ac759b707)
![image](https://github.com/user-attachments/assets/3670608f-46e6-4070-ae9f-c03d0ebee3ef)
![image](https://github.com/user-attachments/assets/b3a8b4bd-8515-40bb-9bf5-c64e8aa37b76)
![frontier-thing](https://github.com/user-attachments/assets/f442fb06-1fb3-4f04-a262-54c645e3be4b)

## Requirements



## Breaking changes

- TickerJobsAvailableEvent has had its unreadable dictionary inside dictionary variables replaced with StationJobInformation

**Changelog**


:cl:
- remove: Removed old NFLatejoin menu
- add: Added new LateJoin menu with more descriptive stations and station and crew separation
